### PR TITLE
Support for ForeignKey

### PR DESCRIPTION
**Pull request by ManabuSeki**
- Support for ForeignKey of MySQL

see document 
- ja
  - https://dev.mysql.com/doc/refman/8.0/ja/create-table-foreign-keys.html
- en
  - https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html

### CODE REVIEW

File: dialect/mysql/mysql.go [hunk 1]
```diff
@@ -300,6 +419,23 @@ func AddPrimaryKey(columns ...string) PrimaryKey {
 	}
 }
 
+// AddForeignKey XXX
+func AddForeignKey(foreignColumns, referenceColumns []string, referenceTableName string, option ...ForeignKeyOption) ForeignKey {
+	foreingKey := ForeignKey{
```
Comment (shmokmt) [inline dialect/mysql/mysql.go]:
```suggestion
	foreignKey := ForeignKey{
```

File: dialect/mysql/mysql.go [hunk 2]
```diff
@@ -261,6 +332,54 @@ func (pk PrimaryKey) ToSQL() string {
...
+func (fk ForeignKey) UpdateOption() string {
+	return fk.updateOption
+}
+
+// DeleteOption XXX
+func (fk ForeignKey) DeleteOption() string {
+	return fk.deleteOption
+}
+
+// ToSQL return foreign key sql string
```
Comment (shmokmt) [inline dialect/mysql/mysql.go:360]:
```suggestion
// ToSQL returns foreign key sql string
```

**Review by shmokmt — CHANGES_REQUESTED** [review]
@ManabuSeki 
thanks for your pull request 😄 
I found some mistakes for typos.
Please fix it 👀

File: dialect/mysql/mysql.go (hunk 2 above)
Comment (ManabuSeki) [reply dialect/mysql/mysql.go:360]: The return value is one

File: dialect/mysql/mysql.go (hunk 1 above)
Comment (ManabuSeki) [reply dialect/mysql/mysql.go]: fixed 107ce6de97ea3ad1750869ba50e8191eaf3e0751

**Review by shmokmt — APPROVED** [review]
thanks!

**Pull request merged**